Sources/MessagingInApp/MessagingInAppImplementation.swift

```diff
@@ -10,15 +10,22 @@ internal class MessagingInAppImplementation: MessagingInAppInstance {
     private var inAppProvider: InAppProvider
     private var eventListener: InAppEventListener?
 
+    private var identifier: String?
+
     init(diGraph: DIGraph) {
         self.logger = diGraph.logger
         self.queue = diGraph.queue
         self.jsonAdapter = diGraph.jsonAdapter
         self.inAppProvider = diGraph.inAppProvider
+        self.identifier = diGraph.profileStore.identifier
```

Comment:
```suggestion
        self.profileStore = diGraph.profileStore
```

Instead of storing the identifier as a property in the `MessagingInAppImplementation` class, I suggest that we store the dependency itself. 

Biggest reason being that it makes the automated tests easier to maintain. Logic inside of class constructors (such as accessing property `.identifier` from profile store dependency) makes test classes/functions more work to setup since you'll need to mock `profileStore.identifier` on every test function, even if that function doesn't need the identifier. I find it less work to only need to mock `profileStore.identifier` in the functions that access `profileStore.identifier`.

Reply:
That's a good point, I would say, not just in the case of "testing" but this also promotes loose coupling between classes. 👍